libraries/AP_NavEKF3/AP_NavEKF3_Measurements.cpp

```diff
@@ -654,7 +654,7 @@ void NavEKF3_core::readGpsData()
         } else {
             // report GPS fix status
             gpsCheckStatus.bad_fix = true;
-            hal.util->snprintf(prearm_fail_string, sizeof(prearm_fail_string), "Waiting for 3D fix");
+            hal.util->snprintf(prearm_fail_string, sizeof(prearm_fail_string), "Waiting for GPS fix %u", gps.status_arm_min());
```

Comment:
Same comment here about reverting this change

Reply:
Done